lib/jazzy/sourcekitten.rb

```diff
@@ -462,34 +478,53 @@ def self.autolink_text(text, doc, root_decls)
         # Traverse children via subsequence components, if any
         link_target = name_traversal(parts, name_root)
 
-        if link_target && link_target.url && link_target.url != doc.url
-          "<code><a href=\"#{ELIDED_AUTOLINK_TOKEN}#{link_target.url}\">" +
-            raw_name + '</a></code>'
+        if link_target &&
+           !link_target.type.extension? &&
+           link_target.url &&
+           link_target.url != doc.url.split('#').first && # Don't link to parent
+           link_target.url != doc.url # Do not link to self
```

Comment:
lol, this is why everyone has love/hate relationships with code linters! I'll change the other one to "Don't" too.